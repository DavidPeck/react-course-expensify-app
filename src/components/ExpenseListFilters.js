import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter , sortByDate , sortByAmount, sortByName , setStartDate, setEndDate } from '../actions/filters';

export class ExpenseListFilters extends React.Component {
state = {
    calendarFocused: null
};

onDatesChange = ({ startDate, endDate}) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
};

onFocusChange = (calendarFocused) => {
    this.setState( () => ({ calendarFocused }));
}

onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
}

onSortChange = (e) => {
    if(e.target.value === 'date') 
        this.props.sortByDate()
    else if (e.target.value === 'amount') 
        this.props.sortByAmount()
    else if (e.target.value === 'desc')
        this.props.sortByName();
}

render() {
    return (
        <div className="content-container">
            <div className="input-group">
                <div className="input-group__item">
                    <input 
                        type="text" 
                        className="text-input"
                        placeholder="Search expenses"
                        value={this.props.filters.text} onChange={this.onTextChange}
                    />
                </div>
                <div className="input-group__item">
                    <select 
                        className="select"
                        value={this.props.filters.sortBy} 
                        onChange={this.onSortChange}>

                        <option value="amount">Amount</option>    
                        <option value="date">Date</option>
                        <option value="desc">Name</option>
                    </select>
                </div>
                <div className="input-group__item">
                    <DateRangePicker
                        startDate={this.props.filters.startDate}
                        startDateId={`filterStartDate`}
                        endDate={this.props.filters.endDate}
                        endDateId={`filterEndDate`}
                        onDatesChange={this.onDatesChange}
                        focusedInput={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        showClearDates={true}
                        numberOfMonths={1}
                        isOutsideRange={() => false}
                    />
                </div>
            </div>
        </div>    
    )
}

};
const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
};

const mapDispatchToProps = (dispatch) => {
    return {
        setTextFilter: (text) => dispatch(setTextFilter(text)),
        sortByDate: () => dispatch(sortByDate()),
        sortByAmount: () => dispatch(sortByAmount()),
        sortByName: () => dispatch(sortByName()),
        setStartDate: (startDate) => dispatch(setStartDate(startDate)),
        setEndDate: (endDate) => dispatch(setEndDate(endDate))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);