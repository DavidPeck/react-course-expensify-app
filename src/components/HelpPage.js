import React from 'react';

const HelpPage = () => (
    <div>
        HHHHHHHHHHHHHEeeeelp!  I really need somebody!
    </div>
);


export default HelpPage;