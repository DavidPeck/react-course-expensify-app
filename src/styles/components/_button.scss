.button {
    background: $blue;
    color: white;   
    border: none; 
    border-radius: 3px;
    display: inline-block;
    font-weight: 300;
    font-size: $font-size-large;
    line-height: 1;
    padding: $s-size;
    text-align: center;
    text-decoration: none;

}

.button--link {
    background: none;
}

.button--grey {
    background: #888;
}