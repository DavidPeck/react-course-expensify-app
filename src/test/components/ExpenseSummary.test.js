import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary'

test('should render ExpenseSummary with 1 expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={871}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={14} expensesTotal={2554946}/>);
    expect(wrapper).toMatchSnapshot();
});