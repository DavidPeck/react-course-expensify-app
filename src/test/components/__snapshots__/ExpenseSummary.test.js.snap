// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totaling 
      <span>
        $8.71
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpenseSummary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        14
      </span>
       
      expenses
       totaling 
      <span>
        $25,549.46
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
